assets/src/edit-story/components/fontPicker/pickerContainer.js

```diff
@@ -0,0 +1,304 @@
+
+  &:hover {
+    background-color: ${({ theme }) => theme.colors.bg.v12};
+  }
+
+  &:focus {
+    background-color: ${({ theme }) => theme.colors.bg.v12};
+    outline: none;
+  }
+`;
+
+const NoItem = styled.span`
+  letter-spacing: ${({ theme }) => theme.fonts.label.letterSpacing};
+  padding: 8px 12px 0 12px;
+  margin: 0;
+  font-style: italic;
+  color: ${({ theme }) => rgba(theme.colors.bg.v0, 0.54)};
+  font-size: ${({ theme }) => theme.fonts.body1.size};
+  line-height: ${({ theme }) => theme.fonts.body1.lineHeight};
+`;
+
+const BoxedTextInput = styled(TextInput)`
+  padding: 6px;
+  border: 1px solid ${({ theme }) => theme.colors.bg.v4};
+  border-radius: 4px;
+  background-color: ${({ theme }) => theme.colors.fg.v1};
+  color: ${({ theme }) => theme.colors.fg.v0};
+`;
+
+const ExpandedTextInput = styled(BoxedTextInput)`
+  flex-grow: 1;
+  margin: 8px;
+`;
+
+function FontPickerContainer({ handleCurrentValue, toggleOptions }) {
+  const {
+    state: { fonts, recentUsedFontSlugs },
+    actions: { addUsedFont, getMenuFonts },
+  } = useFont();
+
+  const pickerContainerRef = useRef();
+  const listContainerRef = useRef();
+  const listRef = useRef();
+  const inputRef = useRef();
+  const currentActiveRef = useRef(0);
+
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [searchValue, setSearchValue] = useState('');
+  const [menuFonts, setMenuFonts] = useState([]);
+
+  // Recently used font options, clean up on refresh
+  const recentUsedFonts = useMemo(
+    () =>
+      searchValue && searchValue !== ''
+        ? recentUsedFontSlugs
+            .map((slug) => fonts.find((font) => font.slug === slug))
+            .filter(({ name }) =>
+              name.toLowerCase().includes(searchValue.toLowerCase())
+            )
+        : recentUsedFontSlugs.map((slug) =>
+            fonts.find((font) => font.slug === slug)
+          ),
+    [searchValue, recentUsedFontSlugs, fonts]
+  );
+
+  // Font options that start with search term
+  const normalFonts = useMemo(
+    () =>
+      searchValue && searchValue !== ''
+        ? fonts.filter(({ name }) =>
+            name.toLowerCase().startsWith(searchValue.toLowerCase())
+          )
+        : fonts,
+    [searchValue, fonts]
+  );
+
+  // Font options that include search term but not start with
+  const includeSearchFonts = useMemo(
+    () =>
+      searchValue && searchValue !== ''
+        ? fonts.filter(
+            ({ name }) =>
+              name.toLowerCase().indexOf(searchValue.toLowerCase()) > 0
+          )
+        : [],
+    [searchValue, fonts]
+  );
+
+  useEffect(() => {
+    const combinedFontList = []
+      .concat(recentUsedFonts)
+      .concat(normalFonts)
+      .concat(includeSearchFonts)
+      .slice(currentIndex, currentIndex + 10)
+      .map(({ name }) => name);
+    getMenuFonts(combinedFontList).then((result) => {
+      const resultArray = result.replace(/}/g, '}},').split('},');
```

Comment:
I would recommend moving this parsing logic to `getMenuFonts`. Also, can you please add some documentation on why we need this?

Reply:
Yes, we don't need this part anymore.